src/core/ScreenLockListenerDBus.cpp

```diff
@@ -0,0 +1,66 @@
+#include "ScreenLockListenerDBus.h"
+
+#include <QDBusConnection>
+#include <QDBusInterface>
+#include <QDBusReply>
+
+ScreenLockListenerDBus::ScreenLockListenerDBus(QWidget *parent):
+    ScreenLockListenerPrivate(parent)
+{
+    QDBusConnection sessionBus = QDBusConnection::sessionBus();
+    QDBusConnection systemBus = QDBusConnection::systemBus();
+    sessionBus.connect(
+                "org.gnome.SessionManager", // service
+                "/org/gnome/SessionManager/Presence", // path
+                "org.gnome.SessionManager.Presence", // interface
+                "StatusChanged", // signal name
+                this, //receiver
+                SLOT(gnomeSessionStatusChanged(uint)));
+
+    systemBus.connect(
+                "org.freedesktop.login1", // service
+                "/org/freedesktop/login1", // path
+                "org.freedesktop.login1.Manager", // interface
+                "PrepareForSleep", // signal name
+                this, //receiver
+                SLOT(logindPrepareForSleep(bool)));
+
+    sessionBus.connect(
+                "com.canonical.Unity", // service
+                "/com/canonical/Unity/Session", // path
+                "com.canonical.Unity.Session", // interface
+                "Locked", // signal name
+                this, //receiver
+                SLOT(unityLocked()));
+
+    /* Currently unable to get the current user session from login1.Manager
```

Comment:
Hi TheZ3ro and Happy new year :)
I would remove the comment and add a separate issue.
In theory the login1.Manager interface should provide a common API for handling session status notifications, however I had issues making it work. when I passed the current pid to the getsession call, it wouldn't return a valid session url. Maybe it only works on consoles but it would be nice if it worked across desktop managers.
Once this issue is resolved, I would de-comment and add a signal listener on the returned session much like what is currently done for unity, gnome session etc.